W-18366625 multimodal support (#71)
mulesoft/docs-idp: modules/ROOT/pages/quotas-and-limits.adoc

--
+
The total number of tokens in a request includes: 
+
** The text contained in the source document to analyze
** The image to analyze with image recognition capabilities
** The sum of all field, table, column names with their corresponding intructions and the schema built through *Customize Schema*
** Approximately 1,000 tokens that IDP uses to process the request
| Prompt Response Limits 
a| 
* 128 characters for IDP NLP prompts
* Approximately 3,000 output words for Einstein prompts using `GPT-4o (gpt-4o-2024-05-13)`
* Approximately 4,000 output words for Einstein prompts using `GPT-4o (gpt-4o-2024-08-06)` or `GPT-4o Mini (gpt-4o-mini-2024-07-18)`
* Approximately 6,000 output words for Einstein prompts using `Einstein GEMINI-2.0 Flash 001`
| Request Limits
a| 
* IDP fails the execution when the number of requests exceeds the Einstein rate limit for the Salesforce Org
** When using Einstein and *Customize Schema* is enabled: 300 requests per minute
** When using Einstein and *Customize Schema* is disabled: (300 / number Einstein of prompts in document action) requests per minute
* 100 concurrent jobs per minute
| Document Languages 
a| 

> Language support varies depending on the selected Large Language Model (LLM) in document actions. 

Author: ```suggestion
Language support varies depending on the selected large language model (LLM) in document actions. 
```

For a list of supported languages by provider, see: 

* https://help.openai.com/en/articles/8357869-how-to-change-your-language-setting-in-chatgpt#h_513834920e[OpenAI's supported languages]
* https://ai.google.dev/gemini-api/docs/models#supported-languages[Gemini's supported languages]
| Polling Limits 
a| 
10-second interval for retrieving results +
Applications must avoid polling for results more often than every 10 seconds. 
|===

[[tokens]] 
== Token Usage and Estimation

> A token is a unit of text that a large language model (LLM) processes. Tokens are essential for breaking down input data into manageable pieces that can be analyzed and generated. These are some estimated conversions that might help understand token lengths: 

Reviewer: ```suggestion
A token is a unit of text that a Large Language Model (LLM) processes. Tokens are essential for breaking down input data into manageable pieces that can be analyzed and generated. These are some estimated conversions that might help understand token lengths: 
```

> A token is a unit of text that a large language model (LLM) processes. Tokens are essential for breaking down input data into manageable pieces that can be analyzed and generated. These are some estimated conversions that might help understand token lengths: 

Author: ```suggestion
A token is a unit of text that an LLM processes. Tokens are essential for breaking down input data into manageable pieces that can be analyzed and generated. These are some estimated conversions that might help understand token lengths: 
```
